src/cryptography/hazmat/bindings/_rust/x509.pyi

```diff
@@ -62,10 +62,26 @@ class PolicyBuilder:
     def time(self, new_time: datetime.datetime) -> PolicyBuilder: ...
     def store(self, new_store: Store) -> PolicyBuilder: ...
     def max_chain_depth(self, new_max_chain_depth: int) -> PolicyBuilder: ...
+    def build_client_verifier(self) -> ClientVerifier: ...
     def build_server_verifier(
         self, subject: x509.verification.Subject
     ) -> ServerVerifier: ...
 
+class ClientVerifier:
+    @property
+    def validation_time(self) -> datetime.datetime: ...
+    @property
+    def store(self) -> Store: ...
+    @property
+    def max_chain_depth(self) -> int: ...
+    def verify(
+        self,
+        leaf: x509.Certificate,
+        intermediates: list[x509.Certificate],
+    ) -> tuple[
+        x509.Name, x509.SubjectAlternativeName | None, list[x509.Certificate]
+    ]: ...
```

Comment:
Before I go further here: does this return type make sense? I figure it makes sense to return the subject explicitly (to make usage clear + make misuse harder, since the types won't align with server verification), but it feels a little funky.

OTOH, I can't think of a cleaner signature -- I think this needs to return both the subject and the SAN, since client verification isn't super well-defined.

Reply:
Named type makes sense! 

Re: subject: in that case, perhaps client verification should enforce the presence of a SAN? That should be trivial to do.